### DISCUSSION TITLE
[Shipped] Optimizing Resources for Web

### DISCUSSION THREAD

**Discussion by zoesyc** [getsentry/sentry#60424] [Launch Feedback]
Hi Everyone - I’m Zoe from the product team 👋 with a third exciting Performance announcement in two weeks!

We are launching a new workflow centred on providing better visibility into resource performance - specifically JavaScript, CSS, and Fonts - across your web application in production. 

Resources - assets like CSS, HTML, JavaScript, Fonts,Images, Videos - are a significant factor in understanding web performance. Poor resources (or asset) performance can come in several flavours, including:

- Uncompressed Assets
- Render-blocking Resources
- Uncached Resources
- Large Assets
- Unoptimized CDN

And often, the result of these instances can lead to poor web vital performance and, ultimately, poor user experience and retention. 

Starting today, you can find **Resources** under **Performance** in the sidebar. 

![Screenshot 2023-11-20 at 5 04 32 PM](https://github.com/getsentry/sentry/assets/38118634/2b65eb20-fd08-4dbc-ab67-33bade322c8a)

<h2>What’s New</h2>

- <h4>Render-blocking Resources </h4> Set "Blocking" filter to ‘Yes’ to quickly identify which scripts, stylesheets, and fonts are blocking core functionality on your page
- <h4>Resource Summary </h4> Pinpoint which pages are blocked or impacted by a given asset with stats like average transfer / encoded / decoded size, average duration, time spent, and requests per minute

<h2>What’s Next</h2>

- **Images and Videos** - Same workflow for more dynamic assets like images and videos
- **Bundle Analysis** - In a collaboration with [CodeCov](https://github.com/codecov/engineering-team/issues/655), we want to help you better optimize your bundles across your product lifecycle
- **Web Vitals** - full integration [our web vitals workflow](https://github.com/getsentry/sentry/discussions/59620) that highlights the best resource optimizations for a better LCP, FCP, and FID score
- **Better Data**
     - Better resource parameterization via bundler plugins 
     - Display cache hit/miss ratio for each resource
     - Display aggregate resource timing info

<h2> We want to hear from you!</h2>
We're rethinking our frontend performance workflows and we'd like to know if this is working for you! We'd like to better understand:
<br></br>

1. Is the Resources workflow intuitive and useful for investigating the performance of your assets?  More specifically, are the resources grouped / parameterized effectively for you to troubleshoot performance problems
2. At what stage in the product life cycle do you most often monitor your resources?
3. What metrics or information would you like to understand about your bundles?
4. Are you interested in tracking resource performance for mobile web instances or mobile apps? What metrics would be most useful in this case?

To be able to use Resources, you'll need to have Performance set up - here's our [setup guide](https://docs.sentry.io/product/performance/getting-started/) for instructions. You can learn more about this release in our [docs](https://docs.sentry.io/product/performance/resources/).

Feel free to share your thoughts in the discussion below - or you can find us on [Discord](https://discord.com/invite/Ww9hbqr) and [X](https://twitter.com/getsentry).

**Comment by stayallive** [comment]
I just noticed this tab became active for me, super interesting but also a little confusing. I'll try to explain.

![image](https://github.com/getsentry/sentry/assets/1090754/042c6b0d-c660-482a-8ffa-d564b3f4115c)

In the screenshot I see a few entries I would expect to be combined or the same and it's not very intuitive to me (at first) why they are not or why they would not be.

For example the first 2 entries I would have expected to be combined. And there is also the grouping from the `woff2` files, the last 2 entries could have been rolled up in the first.

From hovering over the links it seems like they are grouped by project, and clicking through seems to confirm that, however this is not visible in the UI and thus pretty confusing at first 😄 Maybe it's just me but I figured I'd report it anyway 👍

**Comment by zoesyc** [comment]
Update: We're actively improving resources parameterization so you can expect to see both new and old groupings of your assets in the Resources module during this development. 

You can track all our current and upcoming work for Resources in [this GH issue](https://github.com/getsentry/sentry/issues/60482)
